src/wp-includes/class-wp-modules.php

```diff
@@ -0,0 +1,251 @@
+<?php
+/**
+ * Modules API: WP_Modules class.
+ *
+ * Native support for ES Modules and Import Maps.
+ *
+ * @package WordPress
+ * @subpackage Modules
+ */
+
+/**
+ * Core class used to register modules.
+ *
+ * @since 6.5.0
+ */
+class WP_Modules {
```

Comment:
IMO, `WP_Script_Modules` would be a clearer name. The `WP_Modules` name is very generic and "module" could refer to anything, like `WP_Widget`.

Reply:
Let's discuss this [in the PR itself](https://github.com/WordPress/wordpress-develop/pull/5818#issuecomment-1883520368) 🙂